src/index.js

```diff
@@ -51,19 +65,17 @@ export default class VueRouter {
 
     this.app = app
 
-    const { mode, options, fallback } = this
-    switch (mode) {
+    switch (this.mode) {
       case 'history':
-        this.history = new HTML5History(this, options.base)
+        this.history.transitionTo(getLocation(this.history.base))
         break
       case 'hash':
-        this.history = new HashHistory(this, options.base, fallback)
+        this.history.transitionTo(getHash(), () => {
+          window.addEventListener('hashchange', () => {
+            this.history.onHashChange()
```

Comment:
This is what made flow unhappy..
Error log (when you remove the dummy method in the comment above):
```
src/index.js:75
 75:             this.history.onHashChange()
                              ^^^^^^^^^^^^ property `onHashChange`. Property not found in
 75:             this.history.onHashChange()
                 ^^^^^^^^^^^^ AbstractHistory

src/index.js:75
 75:             this.history.onHashChange()
                              ^^^^^^^^^^^^ property `onHashChange`. Property not found in
 75:             this.history.onHashChange()
                 ^^^^^^^^^^^^ HTML5History
```